pandas/core/generic.py

```diff
@@ -5536,6 +5536,16 @@ def __finalize__(
                 object.__setattr__(self, name, getattr(other, name, None))
 
         if method == "concat":
+            # Issue #41828, retain the attrs only if all NDFrame have the same
+            # attrs.
+            attrs = other.objs[0].attrs
```

Comment:
can you instead do an all comparison for the attrs, then do the assignment. this pattern is not super obvious

Reply:
@jreback Thanks for the review. I wonder what do you mean by all comparison?
Something like `all([objs.attrs for objs in other.objs[1:]] == other.objs[0].attrs])`?

Reply:
yep